packages/overlay/src/ActiveOverlay.ts

```diff
@@ -116,6 +117,43 @@ const parentOverlayOf = (el: Element): ActiveOverlay | null => {
     return null;
 };
 
+const getFallbackPlacements = (
+    placement: FloatingUIPlacement
+): FloatingUIPlacement[] => {
+    const fallbacks: FloatingUIPlacement[] = [];
+    fallbacks.push(placement);
+    if (placement.startsWith('left')) {
+        fallbacks.push('bottom');
+        fallbacks.push('top');
+        fallbacks[0] = fallbacks[0].replace(
+            'left',
+            'right'
+        ) as FloatingUIPlacement;
+    } else if (placement.startsWith('right')) {
+        fallbacks.push('bottom');
+        fallbacks.push('top');
+        fallbacks[0] = fallbacks[0].replace(
+            'right',
+            'left'
+        ) as FloatingUIPlacement;
+    } else if (placement.startsWith('top')) {
+        fallbacks.push('left');
+        fallbacks.push('right');
+        fallbacks[0] = fallbacks[0].replace(
+            'top',
+            'bottom'
+        ) as FloatingUIPlacement;
+    } else if (placement.startsWith('bottom')) {
+        fallbacks.push('left');
+        fallbacks.push('right');
+        fallbacks[0] = fallbacks[0].replace(
+            'bottom',
+            'top'
+        ) as FloatingUIPlacement;
+    }
+    return fallbacks;
+};
```

Comment:
Great call on the dictionary. `auto` continues to not be an acceptable value for `FloatingUIPlacement` so while it might do what you are saying it _shouldn't_ ever do that...